src/components/backgroundselector/BackgroundSelectorDirective.js

```diff
@@ -95,11 +97,24 @@ goog.require('ga_topic_service');
             dereg();
           });
 
-          scope.$on('gaTopicChange', function(event, newTopic) {
-            // If the layers config is loaded
-            if (gaLayers.getBackgroundLayers()) {
-              updateBgLayer(newTopic);
-            }
+          gaTopic.getTopics().then(function() {
+            scope.$on('gaLayersChange', function() {
```

Comment:
Yes, I was thinking about a new promise as well (for gaLayers)...

Anyway, AFAIK (to be verified) when gaTopicChange is fired, we already know that the getTopics() promise is resolved. So we can either use gaTopicChange or getTopics() promise. But this does not solve the overall problem about the order of execution.

We could try to create local promises, one (p1) resolved for first gaTopciChange event, another (p2) resolved for first gaLayersChange event, add those topics in $q.all([p1, p2]) construct (see https://docs.angularjs.org/api/ng/service/$q), this would assure that both gaTopicChange and gaLayersChange would have beend called at least once.